src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -119,14 +133,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Protected Overrides Function ExpressionKind() As OperationKind
             Dim rightBinary As BoundBinaryOperator = TryCast(Me.Right, BoundBinaryOperator)
             If rightBinary IsNot Nothing Then
-                If TypeOf rightBinary.Left Is BoundCompoundAssignmentTargetPlaceholder Then
+                If rightBinary.Left.Kind = BoundKind.CompoundAssignmentTargetPlaceholder Then
                     Return OperationKind.CompoundAssignmentExpression
                 End If
             End If
 
             Dim rightOperatorBinary As BoundUserDefinedBinaryOperator = TryCast(Me.Right, BoundUserDefinedBinaryOperator)
             If rightOperatorBinary IsNot Nothing Then
-                If TypeOf rightOperatorBinary.Left Is BoundCompoundAssignmentTargetPlaceholder Then
+                If DirectCast(rightOperatorBinary, IBinaryOperatorExpression).LeftOperand.Kind = OperationKind.PlaceholderExpression Then
```

Comment:
It doesn't look like BoundCompoundAssignmentTargetPlaceholder returns this OperationKind. The fact that we cannot get this line of code right after two attempts shows that we have a test gap, which we are not trying to close. Let's close it, let's test the code that we are writing.